File: desktopclient/src/main/java/com/goexp/galgame/gui/task/TagListTask.java
package com.goexp.galgame.gui.task;

import com.goexp.galgame.common.model.TagType;
import com.goexp.galgame.gui.db.mongo.Query;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.concurrent.Task;

import static com.mongodb.client.model.Sorts.ascending;

public class TagListTask extends Task<ObservableList<TagType>> {

    @Override
    protected ObservableList<TagType> call() {
        return FXCollections.observableArrayList(
                Query.TagQuery.tlp.query()
                        .sort(ascending("order"))
                        .list()
        );
    }
}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/task/TagListTask.scala
package com.goexp.galgame.gui.task

import com.goexp.galgame.common.model.TagType
import com.goexp.galgame.gui.db.mongo.query.TagQuery
import com.mongodb.client.model.Sorts.ascending
import javafx.concurrent.Task

import scala.collection.mutable

class TagListTask extends Task[mutable.Buffer[TagType]] {
  override protected def call =
    TagQuery.tlp.query.sort(ascending("order")).scalaList
}